platform/email/platform-email-impl/src/main/java/org/codice/ddf/platform/email/impl/SmtpClientImpl.java

```diff
@@ -0,0 +1,132 @@
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.email.impl;
+
+import static org.apache.commons.lang.Validate.isTrue;
+import static org.apache.commons.lang.Validate.notEmpty;
+import static org.apache.commons.lang.Validate.notNull;
+
+import java.util.Properties;
+
+import javax.mail.Authenticator;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+import javax.mail.Transport;
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.platform.email.SmtpClient;
+
+/**
+ * Supports unauthenticated connections to a mail server and username/password authenticated connections
+ * over TLS. If the username is not blank and the password is not empty, then auth/tls will be used.
+ *
+ * The methods {@link #createSession()} and {@link #send(Message)} are thread-safe.
+ *
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ */
+public class SmtpClientImpl implements SmtpClient {
+
+    private static final String SMTP_HOST_PROPERTY = "mail.smtp.host";
+
+    private static final String SMTP_PORT_PROPERTY = "mail.smtp.port";
+
+    private static final String SMTP_AUTH_PROPERTY = "mail.smtp.auth";
+
+    private static final String SMTP_START_TLS_ENABLE_PROPERTY = "mail.smtp.starttls.enable";
+
+    private static final String TRUE = "true";
+
+    private static final String FALSE = "false";
+
+    private String hostName;
+
+    private Integer portNumber;
+
+    private String userName;
+
+    private String password;
+
+    /**
+     * Set the username for the email server.
+     *
+     * @param userName may be null or blank
+     */
+    public synchronized void setUserName(String userName) {
+        this.userName = userName;
+    }
+
+    /**
+     * Set the password for the email server.
+     *
+     * @param password may be null or blank
+     */
+    public synchronized void setPassword(String password) {
+        this.password = password;
+    }
+
+    /**
+     * @param hostName must be non-empty
+     */
+    public synchronized void setHostName(String hostName) {
+        notEmpty(hostName, "hostName must be non-empty");
+        this.hostName = hostName;
+    }
+
+    /**
+     * @param portNumber must be non-null and >0
+     */
+    public synchronized void setPortNumber(Integer portNumber) {
+        notNull(portNumber, "portNumber must be non-null");
+        isTrue(portNumber > 0, "portNumber must be >0");
+        this.portNumber = portNumber;
+    }
+
+    @Override
+    public synchronized Session createSession() {
+
+        Properties properties = System.getProperties();
```

Comment:
Since this is initialized from the system properties, I think we should either check for existing values or explicitly state that we will not use them. I could reasonably see someone expect to be able to configure this once in their environment and be done.